osf/management/commands/export_user_account.py

```diff
@@ -0,0 +1,264 @@
+
+ERRORS = []
+
+def export_metadata(node, current_dir):
+    """
+    Exports the pretty printed serialization of a given model instance to metadata.json.
+    Only simple fields (non-FK, non-M2M, etc) are serialized.
+
+    """
+    export_fields = NODE_EXPORT_FIELDS
+    if isinstance(node, Registration):
+        export_fields = REGISTRATION_EXPORT_FIELDS
+    elif isinstance(node, PreprintService):
+        export_fields = PREPRINT_EXPORT_FIELDS
+    with open('{}metadata.json'.format(current_dir), 'w') as f:
+        # only write the fields dict, throw away pk and model_name
+        metadata = serializers.serialize('json', [node], fields=export_fields)
+        f.write(json.dumps(metadata, indent=4, sort_keys=True))
+
+def export_files(node, user, current_dir):
+    """
+    Creates a "files" directory within the current directory.
+    Exports all of the OSFStorage files for a given node.
+    Uses WB's download zip functionality to download osfstorage-archive.zip in a single request.
+
+    """
+    files_dir = '{}files/'.format(current_dir)
+    os.mkdir(files_dir)
+    response = requests.get(
+        url=waterbutler_api_url_for(
+            node_id=node._id,
+            _internal=True,
+            provider='osfstorage',
+            zip='',
+            cookie=user.get_or_create_cookie()
+        )
+    )
+    if response.status_code == 200:
+        with open('{}osfstorage-archive.zip'.format(files_dir), 'wb') as f:
+            f.write(response.content)
+    else:
+        ERRORS.append(
+            'Error exporting files for node {}. Waterbutler responded with a {} status code. Response: {}'
+            .format(node._id, response.status_code, response.json())
+        )
+
+def export_wikis(node, current_dir):
+    """
+    Creates a "wikis" directory within the current directory.
+    Exports all of the wiki pages for a given node as individual markdown files.
+
+    """
+    wikis_dir = '{}wikis/'.format(current_dir)
+    os.mkdir(wikis_dir)
+    for wiki_name, wiki_id in node.wiki_pages_current.iteritems():
+        wiki = NodeWikiPage.objects.get(guids___id=wiki_id)
+        if wiki.content:
+            with io.open('{}{}.md'.format(wikis_dir, wiki_name), 'w', encoding='utf-8') as f:
+                f.write(wiki.content)
+
+def export_node(node, user, current_dir):
+    """
+    Exports metadata, files, and wikis for given node (project, registration, or preprint).
+    If the given node has children,
+        Creates a "components" directory.
+        Recursively exports the node's children.
+        Note: *Sometimes* an empty "components" directory will be created if the given node has children,
+        but the user being exported does not have access to them.
+
+    """
+    export_metadata(node, current_dir)
+    if node.wiki_pages_current:
+        export_wikis(node, current_dir)
+    if OsfStorageFileNode.objects.filter(node=node):
+        export_files(node, user, current_dir)
+
+    if node.nodes:
```

Comment:
Since you're iterating through readable descendants, I think that's what needs to be checked here.

```python
descendants = list(node.find_readable_descendants(...))
if len(descendants):
     ...
```